### PR TITLE
Add workflow to label direct dependencies as priority

### PR DESCRIPTION
**Pull request by patcon**
Resolves: #2175

To close the other lingering draft PR https://github.com/GCTC-NTGC/gc-digital-talent/pull/2176

Source: https://docs.github.com/en/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions#label-a-pull-request

## To Do
- [x] add workflow
- [x] test/debug workflow on fork
- [x] create label in upstream https://github.com/GCTC-NTGC/gc-digital-talent/labels
  - name: `dependencies-priority`
  - color: `#d66103`
  - description: `Pull requests that update a dependency file (top-level package)`

### CODE REVIEW

**Comment by patcon** [comment]
Marking as draft because not part of sprint. Just didn't want to close the other PR without copy-pasting the suggested code into a draft.

**Comment by tristan-orourke** [comment]
Very cool! Let's go ahead and try to merge this, this looks helpful. It also looks pretty simple! Though I'm not exactly sure how to test it...

**Comment by patcon** [comment]
Ah, you're right. I think I can test via:
- push branch to my personal fork
- set the forks default branch to the feature branch
- close all existing dependabot PRs
- force dependabot to check again
- dependabot will create new PRs in the fork, and can see if the workflow works

Checking it here: https://github.com/patcon/gc-digital-talent/pull/424 (currently rebasing after I already found an issue)

**Comment by patcon** [comment]
Ok, this is looking for approval:

Validation that it works is that this issue shows that a special label is added when the issue is a "direct dep". You can see that working at the bottom of this PR:
https://github.com/patcon/gc-digital-talent/pull/424

File: .github/workflows/dependabot-auto-label.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+on: pull_request_target
+
+permissions:
+  pull-requests: write
+  issues: write
+  repository-projects: write
+
+jobs:
+  dependabot:
+    runs-on: ubuntu-latest
```
Comment (petertgiles) [inline .github/workflows/dependabot-auto-label.yml:14]: Don't we prefer pinned versions?

**Review by petertgiles — APPROVED** [review]
Looks cool!  I can see the label on your test branch.  :shrug: Let's try it out.  :laughing:

File: .github/workflows/dependabot-auto-label.yml (hunk 1 above)
Comment (patcon) [reply .github/workflows/dependabot-auto-label.yml:14]: Ack, sorry, i overzealously merged while on the call 🙃

**Pull request merged**